minari/data_collector/data_collector.py

```diff
@@ -98,7 +115,7 @@ def __init__(
         self.max_buffer_steps = max_buffer_steps
 
         # Initialzie empty buffer
-        self._buffer: List[EpisodeBuffer] = [{key: [] for key in STEP_DATA_KEYS}]
+        self._buffer: List[EpisodeBuffer] = [{}]
```

Comment:
Basically, previously because we did not accept unflattened dict observations, all values in a particular episode `dict` could be type `list`. Now, observations and actions can optionally be represented at the top level as a dict, assuming we have a `Dict` space. For example, here's what a buffer looks like in one of our tests:
````
[
    {
        "observations": {
            "component_1": [
                array([-0.62210953], dtype=float32),
                array([-0.9311647], dtype=float32),
                array([-0.3312549], dtype=float32),
                array([-0.62982476], dtype=float32),
                array([0.2710983], dtype=float32),
                array([0.40480512], dtype=float32),
            ],
            "component_2": {
                "subcomponent_1": [
                    array([2.3156273], dtype=float32),
                    array([2.5730186], dtype=float32),
                    array([2.438703], dtype=float32),
                    array([2.4939127], dtype=float32),
                    array([2.0846245], dtype=float32),
                    array([2.6729383], dtype=float32),
                ],
                "subcomponent_2": [
                    array([4.7563176], dtype=float32),
                    array([4.950112], dtype=float32),
                    array([4.5568027], dtype=float32),
                    array([4.48319], dtype=float32),
                    array([4.708479], dtype=float32),
                    array([4.7264085], dtype=float32),
                ],
            },
        },
        "actions": {
            "component_1": [
                array([0.5437195], dtype=float32),
                array([0.6239445], dtype=float32),
                array([-0.29502943], dtype=float32),
                array([-0.14904726], dtype=float32),
                array([-0.57893777], dtype=float32),
            ],
            "component_2": {
                "subcomponent_1": [
                    array([2.7436242], dtype=float32),
                    array([2.3904452], dtype=float32),
                    array([2.14591], dtype=float32),
                    array([2.2326343], dtype=float32),
                    array([2.7800472], dtype=float32),
                ],
                "subcomponent_2": [
                    array([4.7267957], dtype=float32),
                    array([4.544549], dtype=float32),
                    array([4.3176203], dtype=float32),
                    array([4.5739503], dtype=float32),
                    array([4.3437853], dtype=float32),
                ],
            },
        },
        "rewards": [0, 0, 0, 0, 0],
        "terminations": [False, False, False, False, False],
        "truncations": [False, False, False, False, True],
        "seed": 42,
    },
    ...
]
````